### PR TITLE
Fix verify trust-domain for peer authentication

### PR DESCRIPTION
**Pull request by diemtvu**
The authn_filter created for beta applier set `peerIsOptional=false` for all cases in order to use the same one for all mTLS mode. Howerver, this have the side effect that also skip validate trustdomain.

This PR also avoid configuring authn_filter for mTLS principal extraction for gateway, as it (m)TLS for gateway is not set by peer authentication policy.

Issue: https://github.com/istio/istio/issues/21564

### CODE REVIEW

File: pilot/tools/debug/cmd/common.go [hunk 1]
```diff
@@ -0,0 +1,329 @@
+// Copyright 2020 Istio Authors
```
Comment (yangminzhu) [inline pilot/tools/debug/cmd/common.go]: This file is not needed?

File: pilot/pkg/security/authn/v1beta1/policy_applier_test.go [hunk 2]
```diff
@@ -921,8 +920,25 @@ func TestAuthnFilterConfig(t *testing.T) {
 	jwksURI := ms.URL + "/oauth2/v3/certs"
 
 	cases := []testCase{{
-		name:     "no-request-authn-rule",
-		expected: nil,
+		name: "no-request-authn-rule",
```
Comment (yangminzhu) [inline pilot/pkg/security/authn/v1beta1/policy_applier_test.go]: Can we add test cases for using both alpha and beta policies?

File: pilot/pkg/security/authn/v1beta1/policy_applier.go [hunk 3]
```diff
@@ -77,52 +79,97 @@ func (a *v1beta1PolicyApplier) JwtFilter() *http_conn.HttpFilter {
...
 // istio.authentication.v1alpha1.Policy policy, we specially constructs the old filter config to
 // ensure Authn Filter won't reject the request, but still transform the attributes, e.g. request.auth.principal.
 // proxyType does not matter here, exists only for legacy reason.
-func (a *v1beta1PolicyApplier) AuthNFilter(proxyType model.NodeType) *http_conn.HttpFilter {
-	if len(a.processedJwtRules) == 0 && a.consolidatedPeerPolicy == nil {
-		authnLog.Debug("AuthnFilter: RequestAuthentication nor PeerAuthentication (beta policy) not found, fallback to alpha if available")
-		return a.alphaApplier.AuthNFilter(proxyType)
-	}
-	filterConfigProto := convertToIstioAuthnFilterConfig(a.processedJwtRules)
+func (a *v1beta1PolicyApplier) AuthNFilter(proxyType model.NodeType, port uint32) *http_conn.HttpFilter {
```
Comment (yangminzhu) [inline pilot/pkg/security/authn/v1beta1/policy_applier.go]: Could you update the comment for the change?

File: pilot/tools/debug/cmd/common.go (hunk 1 above)
Comment (diemtvu) [reply pilot/tools/debug/cmd/common.go]: yeah, removed.

**Pull request merged**